neuralpp/symbolic/profiler.py

```diff
@@ -0,0 +1,49 @@
+from contextlib import contextmanager
+from time import monotonic
+from typing import Dict
+
+
+class ProfileLog:
+    def __init__(self):
+        self.counter = 0
+        self.time = 0
+
+    def increase_counter(self):
+        self.counter = self.counter + 1
+
+    def add_time_delta(self, delta):
+        self.time = self.time + delta
+
+
+class Profiler:
+    def __init__(self):
+        self._logs: Dict[str, ProfileLog] = {}
+        self._current_section = None
+        self._current_start_time = None
+
+    def reset(self):
+        self._logs: Dict[str, ProfileLog] = {}
+        self._current_section = None
+        self._current_start_time = None
+
+    @contextmanager
+    def profile_section(self, section_name):
```

Comment:
Sort of "context switch" style design for profiling "sections": save the old section and start a new section, stop the timer and add time difference accordingly.